sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/change_feed_utils.py

```diff
@@ -0,0 +1,151 @@
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Internal Helper functions in the Azure Cosmos database change_feed service.
+"""
+
+import warnings
+from datetime import datetime
+from typing import Any, Dict, Tuple
+
+from azure.cosmos._change_feed.change_feed_state import ChangeFeedMode
+
+def add_args_to_kwargs(
+        args: Tuple[Any, ...],
+        kwargs: Dict[str, Any]
+    ) -> None:
+    """Add positional arguments(args) to keyword argument dictionary(kwargs).
+    Since 'query_items_change_feed' method only allows the following 4 positional arguments in the exact order
+    and types, if the order and types don't match, errors will be raised.
+    If the positional arguments are in the correct orders and types, the arguments will be added to keyword arguments.
+
+    4 positional arguments:
+        - str 'partition_key_range_id': [Deprecated] ChangeFeed requests can be executed against specific partition
+            key ranges. This is used to process the change feed in parallel across multiple consumers.
+        - bool 'is_start_from_beginning': [Deprecated] Get whether change feed should start from
+            beginning (true) or from current (false). By default, it's start from current (false).
+        - str 'continuation': e_tag value to be used as continuation for reading change feed.
+        - int 'max_item_count': Max number of items to be returned in the enumeration operation.
+
+    :param args: Positional arguments. Arguments must be in the following order:
+        1. partition_key_range_id
+        2. is_start_from_beginning
+        3. continuation
+        4. max_item_count
+    :type args: Tuple[Any, ...]
+    :param kwargs: Keyword arguments
+    :type kwargs: dict[str, Any]
+    """
+    if len(args) > 4:
+        raise ValueError(f"Too many arguments. Expected: less than 4, but given: {len(args)}")
+
+    if len(args) > 0:
+        key_and_types = [
+            ('partition_key_range_id', str),
+            ('is_start_from_beginning', bool),
+            ('continuation', str),
+            ('max_item_count', int),
+        ]
+        for i, value in enumerate(args):
+            key, expected_type = key_and_types[i]
+
+            if key in kwargs:
+                raise ValueError(f"'{key}' is in both positional and keyword argument list. Please remove one of them.")
```

Comment:
This should be:
`TypeError: foo() got multiple values for argument '{key}'`

Reply:
This logic also doesn't work for detecting a conflict between `is_start_from_beginning` and `start_time`. (I would probably also do that conversion in this method to save you having to do the extra check in the `validate_kwargs` method.